CONTRIBUTING.md

```diff
@@ -101,3 +101,72 @@ e.g. to re-build and run the integration tests, and filter to only the dictionar
 export TEST_AUTH_TOKEN=<YOUR_AUTH_TOKEN>
 npm run build-and-run-tests -- dictionary
 ```
+
+## Testing local changes to the NodeJS SDK
+
+There are cases where you might want to run the examples with changes that you made to the SDK. One common use-case is
+running the [load generator](examples/nodejs/load-gen) for any new configurations, interceptors, or other network related
+changes.
+
+### Laptop/VM
+
+To test any existing or new example from your Laptop or a virtual machine, go to the example directory where it's `package.json` resides
+and install the local version of your SDK through `npm`. For instance, if you want to run the basic example with your changes to the SDK,
+the commands will look like:
+
+```bash
+./scripts/build-nodejs-sdk.sh
+cd examples/nodejs/cache
+npm install ../../../packages/client-sdk-nodejs
+```
+
+Notice that the `package.json` file in the `cache` directory now would link `@gomomento/sdk` dependency to the local
+version in the filesystem:
+
+`"@gomomento/sdk": "file:../../../packages/client-sdk-nodejs"`
+
+Running the examples after this will pick up the local changes made to the SDK. Note that you have to build the SDK each
+time you make a change through `./scripts/build-nodejs-sdk.sh`. If you have not made changes to `packages/core` or
+`packages/common-integration-tests` (local dependencies of the SDK), you can save some time building only the SDK through
+`./scripts/build-package.sh "client-sdk-nodejs"`.
+
+
+### AWS Lambda
+
+To run any existing or new example with your local changes to the SDK on an AWS Lambda environment, we need to follow some
+extra steps. AWS Lambda deployment will fail if we try to follow the same approach as we did on Laptop/VM above. Hence, we
+need to create a tarball of the SDK and point the examples to the tarball as the dependency instead of the SDK directory,
+or it's `npmjs` version.
+
+Since the `packages/client-sdk-nodejs` package takes a local dependency on `packages/core`, directly
+pointing the tarball for the SDK fails as well. In theory, creating a tarball of the `packages/core` package and using that for the SDK
+dependency, and then using the SDK dependency tarball for the examples dependency should work, but it doesn't as the Lambda
+build gets resolution errors.
+
+The steps that worked for one of the contributors was to remove the local `packages/core` directory altogether and use
+the `@gomomento/sdk-core` live `npmjs` [version](https://www.npmjs.com/package/@gomomento/sdk-core) instead.
+For instance, if you want to deploy the lambda `examples/nodejs/lambda-examples/simple-get` to your AWS account with your
+local changes to the SDK, the steps will look like:
+
+1. `rm -rf packages/core` Remove local core directory
```

Comment:
yeah this seems like it shouldn't be necessary but we can explore further and refine this later. this is a great starting point, thanks for doing it